packages-ts/starknet-gauntlet-ledger/src/index.ts

```diff
@@ -0,0 +1,117 @@
+  }
+
+  static create = async (path?: string): Promise<LedgerSigner> => {
+    if (!path) {
+      path = DEFAULT_LEDGER_PATH
+    } else {
+      const match = LEDGER_PATH_REGEX.exec(path)
+      if (!match) {
+        throw new Error(
+          "Provided ledger path doesn't correspond the expected format m/2645'/579218131'/<application>'/<index>'",
+        )
+      }
+    }
+
+    // work around jest reimporting the package
+    // package is only allowed to be imported once
+    const TransportNodeHid = (await import('@ledgerhq/hw-transport-node-hid')).default
+    const transport = await TransportNodeHid.create()
+    const ledgerConnector = new Stark(transport)
+
+    const response = await ledgerConnector.getPubKey(path)
+    if (response.returnCode != LedgerError.NoErrors) {
+      throw new Error(`Unable to get public key: ${response.errorMessage}. Is Ledger app active?`)
+    }
+
+    const publicKey = encode.addHexPrefix(encode.buf2hex(response.publicKey).slice(2, 2 + 64))
+    const signer = new LedgerSigner(ledgerConnector, path, publicKey)
+
+    return signer
+  }
+
+  async getPubKey(): Promise<string> {
+    return this.publicKey
+  }
+
+  async signTransaction(
+    transactions: Invocation[],
+    transactionsDetail: InvocationsSignerDetails,
+    abis?: Abi[],
+  ): Promise<Signature> {
+    const calldata = fromCallsToExecuteCalldataWithNonce(transactions, transactionsDetail.nonce)
+
+    const msgHash = calculcateTransactionHash(
+      transactionsDetail.walletAddress,
+      transactionsDetail.version,
+      getSelectorFromName('__execute__'),
+      calldata,
+      transactionsDetail.maxFee,
+      transactionsDetail.chainId,
+    )
+
+    return this.sign(msgHash)
+  }
+
+  async signMessage(typedData: TypedData, accountAddress: string): Promise<Signature> {
+    const msgHash = getMessageHash(typedData, accountAddress)
+    return this.sign(msgHash)
+  }
+
+  async sign(hash: string): Promise<Signature> {
+    const response = await this.client.sign(this.path, hash, true)
+    if (response.returnCode != LedgerError.NoErrors) {
+      throw new Error(`Unable to sign the message: ${response.errorMessage}`)
+    }
+
+    return [
+      encode.addHexPrefix(encode.buf2hex(response.r)),
+      encode.addHexPrefix(encode.buf2hex(response.s)),
+    ]
+  }
+}
+
+export class Wallet implements IStarknetWallet {
+  wallet: LedgerSigner
+  account: string
+
+  private constructor(signer: LedgerSigner, account?: string) {
+    this.wallet = signer
+    this.account = account
+  }
+
+  static create = async (ledgerPath?: string, account?: string): Promise<Wallet> => {
+    const ledgerSigner: LedgerSigner = await LedgerSigner.create(ledgerPath)
+    return new Wallet(ledgerSigner, account)
+  }
+
+  getPublicKey = async (): Promise<string> => await this.wallet.getPubKey()
+  getAccountAddress = (): string => this.account
+}
+
+export const makeWallet = async (env: Env): Promise<IStarknetWallet> => {
```

Comment:
We shouldn't create objects from env, but from config.
We read config from env, among other sources (file, for example).

See if there are opportunities to clean up some of these types and APIs.

Reply:
This version of gauntlet doesn't have notion of `Config`. `flags` are being converted into `Env` that is used across the app as the source of config parameters. We inherit `makeEnv` wording from previous gauntlet versions. I agree that `Config` should be the final destination for the parameters parsed from different sources (flags, env, files etc.) and suggest to address it in a separate ticket since it's unrelated to the topic of this PR.